internal/core/drand_daemon_dkg_proxy.go

```diff
@@ -75,52 +18,34 @@ func (dd *DrandDaemon) DKGStatus(ctx context.Context, request *drand.DKGStatusRe
 	return dd.dkg.DKGStatus(ctx, request)
 }
 
-func (dd *DrandDaemon) Propose(ctx context.Context, terms *drand.ProposalTerms) (*drand.EmptyResponse, error) {
-	return dd.dkg.Propose(ctx, terms)
-}
-
-func (dd *DrandDaemon) Abort(ctx context.Context, abortDKG *drand.AbortDKG) (*drand.EmptyResponse, error) {
-	beaconID := abortDKG.Metadata.BeaconID
-
-	if !dd.beaconExists(beaconID) {
-		return nil, fmt.Errorf("beacon with ID %s is not running on this daemon", beaconID)
+func (dd *DrandDaemon) Command(ctx context.Context, command *drand.DKGCommand) (*drand.EmptyDKGResponse, error) {
+	if command.Metadata == nil {
+		return nil, errors.New("could not find command metadata to read beaconID")
 	}
-
-	return dd.dkg.Abort(ctx, abortDKG)
-}
-
-func (dd *DrandDaemon) Execute(ctx context.Context, execution *drand.StartExecution) (*drand.EmptyResponse, error) {
-	beaconID := execution.Metadata.BeaconID
+	beaconID := command.Metadata.BeaconID
 
 	if !dd.beaconExists(beaconID) {
 		return nil, fmt.Errorf("beacon with ID %s is not running on this daemon", beaconID)
 	}
 
-	return dd.dkg.Execute(ctx, execution)
+	return dd.dkg.Command(ctx, command)
 }
 
-func (dd *DrandDaemon) Accept(ctx context.Context, proposal *drand.AcceptProposal) (*drand.EmptyResponse, error) {
-	beaconID := proposal.Metadata.BeaconID
-
-	if !dd.beaconExists(beaconID) {
-		return nil, fmt.Errorf("beacon with ID %s is not running on this daemon", beaconID)
+func (dd *DrandDaemon) Packet(ctx context.Context, packet *drand.GossipPacket) (*drand.EmptyDKGResponse, error) {
+	if packet.Metadata == nil {
+		return nil, errors.New("could not find command metadata to read beaconID")
 	}
-
-	return dd.dkg.Accept(ctx, proposal)
-}
-
-func (dd *DrandDaemon) Reject(ctx context.Context, proposal *drand.RejectProposal) (*drand.EmptyResponse, error) {
-	beaconID := proposal.Metadata.BeaconID
+	beaconID := packet.Metadata.BeaconID
 
 	if !dd.beaconExists(beaconID) {
 		return nil, fmt.Errorf("beacon with ID %s is not running on this daemon", beaconID)
 	}
 
-	return dd.dkg.Reject(ctx, proposal)
+	return dd.dkg.Packet(ctx, packet)
 }
 
-func (dd *DrandDaemon) BroadcastDKG(ctx context.Context, packet *drand.DKGPacket) (*drand.EmptyResponse, error) {
-	beaconID := packet.Metadata.BeaconID
+func (dd *DrandDaemon) BroadcastDKG(ctx context.Context, packet *drand.DKGPacket) (*drand.EmptyDKGResponse, error) {
+	beaconID := packet.Dkg.Metadata.BeaconID
```

Comment:
Feels like an easy way to crash all nodes would be to send a `drand.DKGPacket` with nil `Metadata`, correct? 

Please test it &  fix it :P